Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1094 @@
+            $MinimumTlsVersion = $_.MinimumTlsVersion
+            $isMinTlsVersionSetOnCustomDomain = $_.isMinTLSVersionSetOnCustomDomain
+            $isMinTLSVersionSetOnCustomDomainPostRemediation = $_.MinimumTlsVersion
+            
+            $frontdoor | Add-Member -NotePropertyName isMinTLSVersionSetOnCustomDomainPostRemediation -NotePropertyValue $isMinTLSVersionSetOnCustomDomainPostRemediation
+            $frontdoor | Add-Member -NotePropertyName PrevMinimumTlsVersion -NotePropertyValue $PrevMinimumTlsVersion
+
+            Write-Host "Setting minimum required TLS Version for  Front Door :  Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            
+            if (-not [System.Convert]::ToBoolean($isMinTlsVersionSetOnCustomDomain))
+            {
+                try
+                {
+                    Write-Host "Setting minimum required TLS version on custom domain..." -ForegroundColor $([Constants]::MessageType.Info)
+                    Write-Host $([Constants]::SingleDashLine)
+                    
+                    #$resource = Enable-AzFrontDoorCustomDomainHttps -ResourceGroupName $resourceGroupName -FrontDoorName $resourceName -FrontendEndpointName $hostName -MinimumTlsVersion $requiredMinTLSVersion
+                    $secret =  Get-AzFrontDoorCdnSecret -ResourceGroupName $resourceGroupName -ProfileName $resourceName
+                    $secretResoure = New-AzFrontDoorCdnResourceReferenceObject -Id $secret.Id
+                    $updateTlsSetting = New-AzFrontDoorCdnCustomDomainTlsSettingParametersObject -CertificateType $CertificateType -MinimumTlsVersion $requiredMinTLSVersion -Secret $secretResoure
+                    $resource = Update-AzFrontDoorCdnCustomDomain -ResourceGroupName $resourceGroupName -ProfileName $resourceName -CustomDomainName $DomainName -TlsSetting $updateTlsSetting
+
+
+                    if($resource){
+                       $frontdoor.isMinTLSVersionSetOnCustomDomainPostRemediation = $true
+                        $frontdoor.minimumTlsVersion = $requiredMinTLSVersion
+                        $customDomainsRemediated += $frontdoor
+                        Write-Host "Minimum required TLS version for Front Door has been set successfully for " $[$DomainName]  -ForegroundColor $([Constants]::MessageType.Update)
+                        Write-Host $([Constants]::SingleDashLine)
+                    }
+                    else
+                    {
+                        $frontDoorsSkipped += $frontdoor
+                        $logResource = @{}
+                        $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                        $logResource.Add("ResourceName",($_.ResourceName))
+                        $logResource.Add("Reason", "Error while setting the minimum required TLS version. Skipping this Front Door.")
```

Comment:
Again, here are we skipping entire front door or just one endpoint

Reply:
Skipping Front door EP

Reply:
Fixed